python/legate_dataframe/lib/core/column.pyx

```diff
@@ -178,6 +198,52 @@ cdef class LogicalColumn:
         """
         self._handle.offload_to(target_mem)
 
+    def to_array(self, *, writeable=False):
+        """Return a view into the column data.
+
+        This method returns a cupynumeric array viewing the columns data.
+        For non-nullable columns ``cupynumeric.asarray(column)`` should also
+        work directly.
+
+        Parameters
+        ----------
+        writeable : bool
+            By default sets the cupynumeric array not not writeable since
+            columns are normally immutable.  This can be overridden by passing ``True``.
+
+        Returns
+        -------
+        array
+            A 1-d cupynumeric array.
+
+        Raises
+        ------
+        TypeError
+            If the dtype is unsupported by cupynumeric.
+        ValueError
+            If the column has a mask and there are masked values.
+        """
+        # Delay import of cupynumeric.
+        from cupynumeric import asarray
+
+        array = self.get_logical_array(check_dtype=True)
+        if array.num_children != 0:
+            raise TypeError(
+                "cupynumeric doesn't support arrays with children (e.g. strings)."
```

Comment:
Cupynumeric went into a long exception here (not sure it was critical).  So opting to just deal with it explicitly...